src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -175,24 +175,24 @@ public override object Clone()
         }
 
         /// <summary>
-        /// Handler for processing incoming message from node.
+        /// Handler for processing incoming message from the peer.
         /// </summary>
-        /// <param name="node">Node sending the message.</param>
+        /// <param name="peer">Peer sending the message.</param>
         /// <param name="message">Incoming message.</param>
         /// <remarks>
         /// TODO: Fix the exception handling of the async event.
         /// </remarks>
-        private async void AttachedNode_MessageReceivedAsync(NetworkPeer node, IncomingMessage message)
+        private async Task OnMessageReceivedAsync(NetworkPeer peer, IncomingMessage message)
         {
-            this.logger.LogTrace("({0}:'{1}',{2}:'{3}')", nameof(node), node.RemoteSocketEndpoint, nameof(message), message.Message.Command);
+            this.logger.LogTrace("({0}:'{1}',{2}:'{3}')", nameof(peer), peer.RemoteSocketEndpoint, nameof(message), message.Message.Command);
             // TODO: Add exception handling
             // TODO: this should probably be on the mempool scheduler and wrapped in a try/catch
             // async void methods are considered fire and forget and not catch exceptions (typical for event handlers)
             // Exceptions will bubble to the execution context, we don't want that!
 
             try
             {
-                await this.ProcessMessageAsync(node, message).ConfigureAwait(false);
+                await this.ProcessMessageAsync(peer, message).ConfigureAwait(false);
```

Comment:
I don't think we need try catch here anymore now that the async execution is called under the nodes context.
If we need each behaviour to be allowed to crash without crashing the node then the fix needs to be in the caller.

Reply:
try/catch in callbacks is necessary because you MUST NEVER throw to the mechanism that calls your callback
of course unless you want to crash the node, which is fine for exceptions that you are not expect
but every attempt to send message to a peer can throw operation cancelled exception, so at least OCE should be cought in every handler that can send message